# Exponential Decay: Drug Dosage

When a drug is administered intravenously, it reaches its peak concentration in the patient's bloodstream very quickly. After this, the concentration **decays exponentially** - decreasing quickly at first, then more and more slowly. 

## Finding an Exponential Model

Immediately after injection, the concentration of a certain drug in a patient's bloodstream is $1.5\text{ mg}/\text{ml}$ (milligrams per milliliter).

After $4$ hours, a doctor measures the concentration of the drug in the patient's bloodstream again, and finds that it has decreased to $0.25\text{ mg}/\text{ml}$.

Find an exponential model for the concentration of the drug in the patient's blood after $t$ hours. 

---

Recall that an exponentially decaying quantity can be modeled by $$C(t) = C_0 e^{-kt},$$ where $C_0 = C(0)$ is the initial quantity, and $k$ is a positive constant (the rate of decay).

In this problem, we know that $C_0 = 1.5$, so we have $$C(t) = 1.5e^{-kt}.$$ We need to solve for the rate of decay $k$ for this particular drug.

We begin by importing SymPy, so we can use symbolic expressions. We let our variable $t$ and our unknown $k$ both be symbols in SymPy, t and k. Then, we can define concentration $C$ as an expression involving k and t.

In [1]:
from sympy import *

In [2]:
t, k = symbols('t, k')
C = 1.5*exp(-k*t)

To solve for $k$, we can use the fact that the concentration after $4$ hours, $C(4)$, is equal to $0.25\text{ mg}/\text{ml}$. 

We express this equation in SymPy by using the function **Eq()**.

In [3]:
# Store C(4) as C_4
C_4 = C.subs(t,2)
# Equate to the given value of C(4)
Eq(C_4, 0.25)

Eq(1.5*exp(-2*k), 0.25)

Then, to solve the equation for k, we use the function **solveset()**.

We add the argument **domain=S.Reals** to ensure that we get only real-number solutions.

In [4]:
solveset(Eq(C_4, 0.25), k, domain=S.Reals)

FiniteSet(0.895879734614027)

Thus, the rate of decay for this drug is $k = 0.895879734614027$.

In [5]:
# Store the value of k
k = 0.895879734614027 

## Calculating Dosage

Most drugs become ineffective if they are below a minimum concentration $m$, and become dangerous if they are above a maximum concentration $M$. That is, the doctor must make sure that $$m \leq C(t) \leq M$$ at all times $t$.

For the same drug discussed above, the minimum effective concentration is $m = 0.6\text{ mg}/\text{ml}$, and the maximum safe concentration is $M = 5\text{ mg}/\text{ml}$.

Suppose an initial dose of the drug results in a concentration of $4\text{ mg}/\text{ml}$.

1. How long can the doctor wait until administering another dose so that the drug remains effective? 
2. A second dose is given after 1 hour. What is the maxium concentration that can be added by this dose?

---

We have the same drug, so we use the same rate of decay, $k = 0.895879734614027$. However, we have a new initial concentration. Thus we can write a new function for the concentration after $t$ hours in this case: $$C_1(t) = 4e^{-kt}.$$

In [6]:
# Store the minimum and maximum concentration
m = 0.6
M = 5
# Define the concentration function
C1 = 4*exp(-k*t)
C1

4*exp(-0.895879734614027*t)

### Time for Minimum Effective Concentration

For the first question, we want to find how long it takes for the concentration to reach the minimum effective level, $0.6\text{ mg}/\text{ml}$. That is, we want to solve for $t$ such that $C_1(t) = m = 0.6$.

We use the **solveset()** function again.

In [7]:
solveset(Eq(C1, m), t, domain=S.Reals)

FiniteSet(2.1176056468151)

The doctor can wait up to approximately $2.12$ hours before administering another dose to the patient. If they wait longer, then the drug will become ineffective.

### Dose for Maximum Safe Concentration

For the second question, we look at the concentration of the drug in the patient's bloodstream after 1 hour, $C_1(1)$.

In [8]:
# Store C1(1) as C1_1
C1_1 = C1.subs(t, 1)
C1_1

1.63299316185545

By how many $\text{mg}/\text{ml}$ can the concentration be increased before the drug becomes dangerous for the patient?

In [9]:
# Difference between maximum concentration and concentration after 1 hour
M - C1_1

3.36700683814455

Thus, the doctor must measure the dose so that it will not result in a concentration greater than $3.367\text{ mg}/\text{ml}$.

# Reference

Annamalai, C. (2010). Applications of exponential decay and geometric series in effective medicine dosage. *Advances in Bioscience and Biotechnology, 1*, 51-54. https://doi.org/10.4236/abb.2010.11008